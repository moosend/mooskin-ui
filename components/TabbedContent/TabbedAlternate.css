@import '../_utils/globals/variables.css';
@import '../_utils/globals/screens.css';

.containerH {
    display: flex;
    flex-direction: column;
}

.containerV {
    display:flex;
    flex-direction: row;
}

.headingV {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    flex-shrink: 1; 
    border-radius: 8px;
    background: #efefef;
    /* border-right: 1px solid var(--moosendBlue); */
}

/* .headingV > .header + .header{
    margin-top: 10px;
} */

.headingH {
    display: flex;
    flex-grow: 1;
    flex-shrink: 1;
    flex-direction: row;
    border-radius: 8px;
    background: #efefef;
    /* border-bottom: 1px solid var(--moosendBlue); */
}

/* .headingH > .header + .header{
    margin-left: 10px;
} */

.content{
    color: #7b7b7b;
    font-size: 0.875rem;
}

.contentH {
    flex-grow: 1;
    flex-shrink: 1;
}

.contentV {
    flex: 2;
}

.header {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    border-radius: 10px;
    /* color: var(--moosendBlue); */
    color: #aaa;
    font-size: 1rem;
    background: transparent;
    transition: background-color 0.3s;
    /* border: 1px solid var(--moosendBlue); */
    padding: 13px;
    cursor: pointer;
    user-select: none;
}

/* .header:hover {
    background-color: var(--radioHover);
    color: #fff;
    border-color: var(--radioHover);
} */

/* .header input[type='radio'] {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 1.125rem;
    height: 1.125rem;
    appearance: none;
    margin-right: 10px;
    cursor: pointer;
    background: #fff;
    border-radius:50%;
    border: 1px solid var(--moosendBlue);
} */

/* .header:hover input[type='radio']{
    border-color: var(--radioHover);
} */

/* .header input[type='radio']:checked::after {
    background: var(--moosendBlue);
    content:'';
    display: block;
    width: 0.375rem;
    height: 0.375rem;
    border-radius:50%; 
} */

.header label {
    display: flex;
    justify-content: center;
    flex-direction: column;
    cursor: pointer;
    font-size: 0.875rem;
    width: 100%;
    height: 100%;
}

/* .header span {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    cursor: pointer;
    user-select: none;
} */

.activeHeader {
    background: var(--moosendBlue);
    color: #fff;
    margin: -3px 0;
}

.activeHeader:hover {
    background: var(--moosendBlue);
    border-color: var(--moosendBlue);
}

/* .activeHeader:hover input[type='radio']{
    border-color: var(--moosendBlue);
} */

.visible {
    display: block;
}

.invisible {
    display: none;
}

@media (--xSmall){

    .headingV{
        margin: 0;
        padding: 0;
        border-right: none;
        margin-bottom: 10px;
        padding-bottom: 10px;
        border-bottom: 1px solid var(--moosendBlue);
    }

    .headingH{
        flex-direction: column;
    }

    /* .headingH > .header + .header{
        margin: 10px 0 0 0
    } */

    .containerV{flex-direction: column}

} 

/* @media (--small){

    .headingV{
        margin: 0;
        padding: 0;
        border-right: none;
        margin-bottom: 10px;
        padding-bottom: 10px;
        border-bottom: 1px solid var(--moosendBlue);
    }

    .headingH{
        flex-direction: column;
    }

    .headingH > .header + .header{
        margin: 10px 0 0 0
    }

    .containerV{flex-direction: column}

} */
