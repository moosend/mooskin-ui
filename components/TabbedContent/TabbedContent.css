@import url('https://fonts.googleapis.com/css?family=Convergence');
@import url('https://fonts.googleapis.com/icon?family=Material+Icons');
@import '../_utils/globals/variables.css';
@import '../_utils/globals/screens.css';

.container{
    display: flex;
    width: 100%;
}

.containerV {
    flex-direction: row;
}

.containerH {
    flex-direction: column;
}

.heading {
    display: flex;
    flex-direction: row;
}

.headingH {
    flex: 1;
    border-bottom:10px solid #efefef;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
}

.headingV {
    flex-direction: column;
    flex: 1;
    border-right:10px solid #efefef;
    border-top-left-radius: 5px;
    border-bottom-left-radius: 5px;
}

.contentH {
    flex: 1;
}

.contentV {
    flex: 2;
}

.header {
    padding: 30px 0;
    flex-grow: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    transition: all 400ms ease;
    border-top-left-radius: inherit;
    border-top-right-radius: inherit;
    border-bottom-left-radius: inherit;
    color: #575757;
    font-family: Convergence,sans-serif;
    font-size: 1.125rem;
}

.headerAlign{
    padding: 0;
    flex-grow: 0;
    width: 120px;
    height: 120px;
}

.headingHRight{
    flex-direction: row-reverse;
}

.headingVRight{
    flex-direction: column-reverse;
}

.headingCenter{
    justify-content: center;
}

.headingVAlign{
    flex: 0;
} 

/* .headerAlign span.title{
    color: #6b6b6b;
    font-family: sans-serif;
    font-size: .92857142857rem;
    border-bottom: 1px solid #efefef;
}

.headerAlign:hover span.title{
    color: var(--moosendBlue);
} */

.tab{
    display: block;
}

.activeHeader {
    background: var(--inputBackground);
}

.inactiveHeader {
    background: #fff;
}

.inactiveHeader:hover {
    background: rgba(245, 245, 245, 0.5);
}

.visible {
    display: block;
}

.invisible {
    display: none;
}

.image{
    margin-top: 10px;
    width: 40px;
}

@media (--deviceSmall){

    .heading{
        height: 10vw;
    }

    .heading span{
        font-size: 4vw;
    }

}

@media (--xSmall){
    .containerV{flex-direction: column}

    .headingV{
        flex-direction: row;
        border-bottom:10px solid #efefef;
        border-right: none;
    }

}
/* 
@media (--small){
    .containerV{flex-direction: column}

    .headingV{
        flex-direction: row;
        border-bottom:10px solid #efefef;
        border-right: none;
    }
} */

@media (--deviceXsmall){

    .heading{
        font-size: 3.5vw;
    }

    .contentH {
        font-size: 3vw;
    }

    .contentV {
        font-size: 3vw;
    }

    .headerAlign{
        flex-grow: 1;
    }

    .header{
        flex: 1;
        font-size: 4vw
    }

}

@media (--deviceSmall){

    .heading{
        font-size: 3.5vw;
    }

    .contentH {
        font-size: 3vw;
    }

    .contentV {
        font-size: 3vw;
    }

    .headerAlign{
        flex-grow: 1;
    }

    .header{
        flex: 1;
        font-size: 4vw
    }

}