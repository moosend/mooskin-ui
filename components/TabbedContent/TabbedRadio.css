@import '../_utils/globals/variables.css';
@import '../_utils/globals/screens.css';

.containerH {
    display: flex;
    flex-direction: column;
}

.containerV {
    display:flex;
    flex-direction: row;
}

.headingV {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    flex-shrink: 1; 
    margin-right: 10px;
}

.borderV {
    border-right: 1px solid var(--moosendBlue);
}

.headingV > .header + .header{
    margin-top: 10px;
}

.headingH {
    display: flex;
    flex-grow: 1;
    flex-shrink: 0;
    flex-direction: column;
    margin-bottom: 10px;
}

.headingH > .header + .header{
    margin: 10px 0px 0px 0px
}

.content{
    color: #7b7b7b;
    font-size: 0.875rem;
}

.contentH {
    flex-grow: 1;
    flex-shrink: 1;
}

.contentV {
    flex: 2;
}

.header {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    width: 100%;
    border-radius: 10px;
    font-size: 0.875rem;
    padding: 15px;
    cursor: pointer;
    user-select: none;
    color: #7b7b7b;
    background: #f2f2f2;
}

.header input[type='radio'] {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 22px;
    height: 22px;
    appearance: none;
    margin-right: 10px;
    cursor: pointer;
    background: #fff;
    border-radius:50%;
}

.header:hover input[type='radio']{
    border-color: var(--radioHover);
}

.header input[type='radio']:checked::after {
    background: #333;
    content:'';
    display: block;
    width: 0.6rem;
    height: 0.6rem;
    border-radius:50%;
}

.header label {
    display: flex;
    justify-content: center;
    flex-direction: column;
    cursor: pointer;
    font-size: 0.875rem;
    width: 100%;
    height: 100%;
}

.header span {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    cursor: pointer;
    user-select: none;
}

.activeHeader {
    background: var(--moosendBlue);
    color: #fff;
}

.activeHeader:hover {
    background: var(--moosendBlue);
    border-color: var(--moosendBlue);
}

.activeHeader:hover input[type='radio']{
    border-color: var(--moosendBlue);
}

.visible {
    display: block;
}

.invisible {
    display: none;
}

@media (--xSmall){

    .headingV{
        margin: 0;
        padding: 0;
        border-right: none;
        margin-bottom: 10px;
    }

    .headingH{
        flex-direction: column;
    }

    .headingH > .header + .header{
        margin: 10px 0 0 0
    }

    .containerV{flex-direction: column}

} 

@media (--small){

    .headingV{
        margin: 0;
        padding: 0;
        border-right: none;
        margin-bottom: 10px;
    }

    .headingH{
        flex-direction: column;
    }

    .headingH > .header + .header{
        margin: 10px 0px 0px 0px
    }

    .containerV{flex-direction: column}

}
