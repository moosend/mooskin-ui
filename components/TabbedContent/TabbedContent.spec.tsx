import * as React from 'react';
import TabbedContent, {Content, Header, Tab} from './TabbedContent';

import { mount, render, shallow } from 'enzyme';

describe('TabbedContent', () => {

    test('renders TabbedContent properly according to snapshot', () => {

        const component = mount(
            <TabbedContent
                id="5"
                className="mySwitch"
                style={{color: 'black'}}
                type="normal"
                vertical
                alignHeaders="right"
            >
                <Tab>
                    <Header className="mySwitch" style={{color: 'black'}}>Moosend</Header>
                    <Content>Table</Content>
                </Tab>
                <Tab active>
                    <Header className="mySwitch" style={{color: 'black'}}>Home</Header>
                    <Content>Some Content</Content>
                </Tab>
                <Tab>
                    <Header className="mySwitch" style={{color: 'black'}}>About</Header>
                    <Content>Blla blla</Content>
                </Tab>
            </TabbedContent>
        );

        expect(component.find('Header').length).toBe(3);
        expect(component.find('Content').length).toBe(3);

        expect(component.find('Header').first().prop('children')).toBe('Moosend');

        expect(component).toMatchSnapshot();
    });

    test('renders TabbedContent vertically with radio styles', () => {

        Math.random = jest.fn(() => 222333444555);

        const component = shallow(
            <TabbedContent vertical type="radio">
                <Tab>
                    <Header className="mySwitch" style={{color: 'black'}}>Moosend</Header>
                    <Content>Table</Content>
                </Tab>
                <Tab active>
                    <Header className="mySwitch" style={{color: 'black'}}>Home</Header>
                    <Content>Some Content</Content>
                </Tab>
                <Tab>
                    <Header className="mySwitch" style={{color: 'black'}}>About</Header>
                    <Content>Blla blla</Content>
                </Tab>
            </TabbedContent>
        );

        expect(component.prop('type')).toBeTruthy;
        expect(component.prop('vertical')).toBeTruthy;

        expect(component).toMatchSnapshot();
    });

    test('renders Tab properly according to snapshot', () => {
        const component = shallow(
            <Tab style={{color: 'blue'}} active/>
        );

        expect(component).toMatchSnapshot();
    });

    test('renders Header properly according to snapshot', () => {
        const component = shallow(
            <Header
                onClick={() => null}
                active
            >
                just some Header
            </Header>
        );

        expect(component).toMatchSnapshot();
    });

    test('renders input and callback function is called when header is clicked', () => {

        const func = jest.fn();
        Math.random = jest.fn(() => 222333444555);

        const component = shallow(
            <TabbedContent type="radio">
                <Tab>
                    <Header onClick={func} className="mySwitch" style={{color: 'black'}}>Moosend</Header>
                    <Content>Table</Content>
                </Tab>
            </TabbedContent>
        );

        component.find(Header).simulate('click');

        expect(component.find('input')).toBeTruthy;
        expect(func).toHaveBeenCalled();

    });

    // TODO: failing test
    test('changes the state and displays appropriate tab when one of the headers is clicked and back', () => {

        const component = mount(
            <TabbedContent>
                <Tab>
                    <Header>Moosend</Header>
                    <Content>Table</Content>
                </Tab>
                <Tab active>
                    <Header>Home</Header>
                    <Content>Some Content</Content>
                </Tab>
            </TabbedContent>
        );

        expect(component.state('activeTab')).toBe(1);
        expect(component.find('.tab-header').first().hasClass('inactiveHeader')).toBe(true);
        expect(component.find('.tab-header').last().hasClass('activeHeader')).toBeTruthy();
        expect(component.find('.content-component').first().hasClass('invisible')).toBeTruthy();
        expect(component.find('.content-component').last().hasClass('visible')).toBeTruthy();

        component.find('Header').first().simulate('click');

        expect(component.state('activeTab')).toBe(0);
        expect(component.find('.tab-header').last().hasClass('inactiveHeader')).toBeTruthy();
        expect(component.find('.tab-header').first().hasClass('activeHeader')).toBeTruthy();
        expect(component.find('.content-component').last().hasClass('invisible')).toBeTruthy();
        expect(component.find('.content-component').first().hasClass('visible')).toBeTruthy();

        component.find('Header').last().simulate('click');

        expect(component.state('activeTab')).toBe(1);
        expect(component.find('.tab-header').first().hasClass('inactiveHeader')).toBeTruthy();
        expect(component.find('.tab-header').last().hasClass('activeHeader')).toBeTruthy();
        expect(component.find('.content-component').first().hasClass('invisible')).toBeTruthy();
        expect(component.find('.content-component').last().hasClass('visible')).toBeTruthy();

    });

    // test('renders an aligned heading with headers & image, headerInfo and headerValue props', () => {

    //     const component = mount(
    //         <TabbedContent align>
    //             <Tab image="imageSrc" title="title1">asdasd1</Tab>
    //             <Tab headerInfo="blla" headerValue={1} title="title2">asdasd2</Tab>
    //         </TabbedContent>
    //     );

    //     expect(component.find('.tab-header').first().hasClass('headerAlign')).toBeTruthy;
    //     expect(component.find('Header').first().prop('image')).toEqual('imageSrc');
    //     expect(component.find('Header').last().prop('info')).toEqual('blla');
    //     expect(component.find('Header').last().prop('value')).toEqual(1);

    // });

    // test('some props should not render when Radio prop is passed', () => {

    //     const component = mount(
    //         <TabbedContent radio>
    //             <Tab materialIcon="face" iconClass="myClass" image="imageSrc" title="title1">asdasd1</Tab>
    //             <Tab headerInfo="blla" headerValue={1} title="title2">asdasd2</Tab>
    //         </TabbedContent>
    //     );

    //     expect(component.find('.tab-header').first().hasClass('headerAlign')).toBeFalsy;
    //     expect(component.find('.material-icons')).toBeFalsy;
    //     expect(component.find('.header-icon')).toBeFalsy;
    //     expect(component.find('.title')).toBeFalsy;
    //     expect(component.find('.info')).toBeFalsy;
    //     expect(component.find('.image')).toBeFalsy;
    // });
});
