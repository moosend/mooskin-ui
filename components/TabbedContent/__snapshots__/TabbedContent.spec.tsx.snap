// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabbedContent renders Header properly according to snapshot 1`] = `
<div
  className="tab-header header activeHeader "
  onClick={[Function]}
  style={
    Object {
      "width": undefined,
    }
  }
>
  just some Header
</div>
`;

exports[`TabbedContent renders Tab properly according to snapshot 1`] = `
<div
  className="tab-component undefined"
  style={
    Object {
      "color": "blue",
    }
  }
/>
`;

exports[`TabbedContent renders TabbedContent properly according to snapshot 1`] = `
<TabbedContent
  activeTab={0}
  alignHeaders="right"
  className="mySwitch"
  id="5"
  style={
    Object {
      "color": "black",
    }
  }
  type="normal"
  vertical={true}
>
  <div
    className="tabbed-content-component container containerV"
    id="5"
  >
    <div
      className="heading headingV headingVRight headingVAlign"
    >
      <Header
        active={false}
        align="right"
        className="mySwitch"
        key="0"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
          }
        }
        type="normal"
        vertical={true}
      >
        <div
          className="tab-header header inactiveHeader headerAlign"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
              "width": undefined,
            }
          }
        >
          Moosend
        </div>
      </Header>
      <Header
        active={true}
        align="right"
        className="mySwitch"
        key="1"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
          }
        }
        type="normal"
        vertical={true}
      >
        <div
          className="tab-header header activeHeader headerAlign"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
              "width": undefined,
            }
          }
        >
          Home
        </div>
      </Header>
      <Header
        active={false}
        align="right"
        className="mySwitch"
        key="2"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
          }
        }
        type="normal"
        vertical={true}
      >
        <div
          className="tab-header header inactiveHeader headerAlign"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
              "width": undefined,
            }
          }
        >
          About
        </div>
      </Header>
    </div>
    <div
      className="content contentV"
    >
      <Content
        active={false}
        key="0"
      >
        <div
          className="content-component invisible undefined"
          style={undefined}
        >
          Table
        </div>
      </Content>
      <Content
        active={true}
        key="1"
      >
        <div
          className="content-component visible undefined"
          style={undefined}
        >
          Some Content
        </div>
      </Content>
      <Content
        active={false}
        key="2"
      >
        <div
          className="content-component invisible undefined"
          style={undefined}
        >
          Blla blla
        </div>
      </Content>
    </div>
  </div>
</TabbedContent>
`;

exports[`TabbedContent renders TabbedContent vertically with radio styles 1`] = `
<div
  className="tabbed-content-component container containerV"
>
  <div
    className="heading headingV headingCenter "
  >
    <Header
      active={false}
      className="mySwitch"
      key="0"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
      type="radio"
      vertical={true}
    >
      Moosend
    </Header>
    <Header
      active={true}
      className="mySwitch"
      key="1"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
      type="radio"
      vertical={true}
    >
      Home
    </Header>
    <Header
      active={false}
      className="mySwitch"
      key="2"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
      type="radio"
      vertical={true}
    >
      About
    </Header>
  </div>
  <div
    className="content contentV"
  >
    <Content
      active={false}
      className=""
      key="0"
      style={Object {}}
    >
      Table
    </Content>
    <Content
      active={true}
      className=""
      key="1"
      style={Object {}}
    >
      Some Content
    </Content>
    <Content
      active={false}
      className=""
      key="2"
      style={Object {}}
    >
      Blla blla
    </Content>
  </div>
</div>
`;
